core/domain/collection_domain.py

```diff
@@ -693,6 +865,29 @@ def validate(self, strict=True):
         for node in self.nodes:
             node.validate()
 
+        for skill_id, skill in self.skills.iteritems():
```

Comment:
What about also validating skill name uniqueness and consistency between the skills list and ``prerequisite_skill_ids`` and ``acquired_skill_ids`` for each collection node?